ToDoListAPI/ToDoListAPI/Services/UserService.cs

```diff
@@ -0,0 +1,131 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Logging;
+using Microsoft.IdentityModel.Tokens;
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using ToDoListAPI.IService;
+using ToDoListAPI.Models;
+
+namespace ToDoListAPI.Services
+{
+    public class UserService : IUserService
+    {
+        private MyDBContext _context;
+        private IConfiguration _configuration;
+
+        public UserService(MyDBContext context, IConfiguration configuration)
+        {
+            _context = context;
+            _configuration = configuration;
+        }
+
+        public async Task<List<User>> GetAll()
+        {
+            return await _context.Users.ToListAsync();
+        }
+
+        public async Task<string> CreateUser(User user)
+        {
+            //Kiem tra user name duy nhat
+            var checkUserName = await _context.Users.FirstOrDefaultAsync(u => u.UserName == user.UserName);
+            //Khác null nghĩa là đã tồn tại
+            if (checkUserName != null)
+            {
+                return "01";
+            }
+            //Chưa tồn tại thì thêm
+            else
+            {
+                _context.Users.Add(user);
+                await _context.SaveChangesAsync();
+                return "00";
+            }
+        }
+
+        public async Task<string> Login(string username, string pass)
+        {
+            var res = "";
+            var user = await _context.Users.FirstOrDefaultAsync(user => user.UserName == username && user.Password == pass);
+            if (user == null)
+            {
+                res = "01";
+            }
+            else
+            {
+                //GenerateToken token
+
+                res = GenerateToken(user);
+            }
+
+            return res;
+        }
+
+        //Kiểm tra token hợp lệ
+        public string VerifyToken(string token)
+        {
+            IdentityModelEventSource.ShowPII = true;
+
+            SecurityToken validatedToken;
+            TokenValidationParameters validationParameters = new TokenValidationParameters();
+
+            try
+            {
+                validationParameters.ValidateLifetime = true;
+
+                validationParameters.ValidAudience = _configuration["Jwt:Issuer"].ToLower();
+                validationParameters.ValidIssuer = _configuration["Jwt:Issuer"].ToLower();
+                validationParameters.IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_configuration["Jwt:Key"]));
+
+                ClaimsPrincipal principal = new JwtSecurityTokenHandler().ValidateToken(token, validationParameters, out validatedToken);
+                return "00";
+            }
+            catch
+            {
+                return "01";
+            }
+        }
```

Comment:
phần verify jwt để dùng được `[Authorize]` e tham khảo link dưới nhé
https://dev.to/eduardstefanescu/jwt-authentication-with-symmetric-encryption-in-asp-net-core-2i53